subsys/nrf_security/src/drivers/cracen/cracenpsa/src/ctr_drbg.c

```diff
@@ -116,7 +116,7 @@ static psa_status_t ctr_drbg_update(uint8_t *data)
 }
 
 /* Instantiation of CTR_DRBG without derivation function. */
-psa_status_t cracen_init_random(cracen_prng_context_t *context)
+static psa_status_t cracen_init_random_internal(cracen_prng_context_t *context)
```

Comment:
I thought locking a mutex twice would have been okay. :thinking:

Reply:
So I thought that you were joking because this seems like contradictive by definition but you are actually right for Zephyr. The mutex implementation of Zephyr actually supports the same thread to lock the mutex multiple times. So I will revert the commit and just add a comment.